tests/plugins/imagebase/features/caption.js

```diff
@@ -0,0 +1,334 @@
+/* bender-tags: editor,widget */
+/* bender-ckeditor-plugins: imagebase,toolbar */
+/* bender-include: ../../widget/_helpers/tools.js */
+/* global widgetTestsTools */
+
+( function() {
+	'use strict';
+
+	bender.editors = {
+		classic: {},
+
+		divarea: {
+			config: {
+				extraPlugins: 'divarea'
+			}
+		},
+
+		inline: {
+			creator: 'inline'
+		}
+	};
+
+	function addTestWidget( editor ) {
+		if ( editor.widgets.registered.testWidget ) {
+			return;
+		}
+
+		var plugin = CKEDITOR.plugins.imagebase;
+
+		plugin.addImageWidget( editor, 'testWidget', plugin.addFeature( editor, 'caption', {} ) );
+	}
+
+	function getFixture( name ) {
+		return CKEDITOR.document.getById( name ).getHtml();
+	}
+
+	function assertVisibility( caption, isVisible, msg ) {
+		assert[ 'is' + ( isVisible ? 'False' : 'True' ) ]( !!caption.data( 'cke-hidden' ), msg );
+	}
+
+	function createToggleTest( options ) {
+		return function( editor, bot ) {
+			addTestWidget( editor );
+
+			bot.setData( getFixture( options.fixture ), function() {
+				var widget = widgetTestsTools.getWidgetByDOMOffset( editor, 0 ),
+					caption = widget.parts.caption,
+					focusTrap = editor.editable().findOne( 'p' ).getChild( 0 ),
+					range = editor.createRange();
+
+				assertVisibility( caption, options.initial, 'caption visibility (initial)' );
+
+				if ( options.onInit ) {
+					options.onInit( widget );
+				}
+
+				if ( options.customFocus ) {
+					options.customFocus( widget );
+				} else {
+					widget.focus();
+				}
+
+				assertVisibility( caption, options.focus, 'caption visibility (focus)' );
+
+				if ( options.onFocus ) {
+					options.onFocus( widget );
+				}
+
+				range.setStart( focusTrap, 1 );
+				range.collapse();
+				range.select();
+
+				assertVisibility( caption, options.blur, 'caption visibility (blur)' );
+
+				if ( options.onBlur ) {
+					options.onBlur( widget );
+				}
+			} );
+		};
+	}
+
+	function createMultipleToggleTest( options ) {
+		return function( editor, bot ) {
+			var forEach = CKEDITOR.tools.array.forEach;
+
+			addTestWidget( editor );
+
+			function getWidgets( editor, amount ) {
+				var i,
+					widgets = [];
+
+				for ( i = 0; i < amount; i++ ) {
+					widgets.push( widgetTestsTools.getWidgetByDOMOffset( editor, i ) );
+				}
+
+				return widgets;
+			}
+
+			function assertMultipleVisibility( widgets, expected, msg ) {
+				forEach( widgets, function( widget, i ) {
+					assertVisibility( widget.parts.caption, expected[ i ],
+						'caption#' + i + ' visibility (' + msg + ')' );
+				} );
+			}
+
+			function assertOnFocus( widgets, expected ) {
+				forEach( widgets, function( widget, i ) {
```

Comment:
In reality this code will never execute assertions for more than 1 widget. In YUI `wait` throws an exception, so the flow is  never resumed afterwards.